ballerina/modules/parser/lexer.bal

```diff
@@ -103,22 +103,112 @@ public class Lexer {
+                    return error UnterminatedStringError(message, line = self.currentLocation.line,
+                                                         column = self.currentLocation.column);
+                } else {
+                    value = self.readNextChar();
+                    if value == DOUBLE_QUOTE && previousChar != BACK_SLASH {
+                        break;
+                    } else {
+                        word += value;
+                    }
+                    previousChar = value;
+                }
+            }
+            return getToken(word, T_STRING, location);
+        }
+    }
+
+    isolated function readBlockStringLiteral() returns Token|SyntaxError {
+        string previousChar = "";
+        string word = "";
+        string[] lines = [];
+        Location location = self.currentLocation.clone();
+        _ = self.readNextChar(); // Ignore third double quote character
+        while true {
             string value = self.charReader.peek();
-            if (value is LineTerminator) {
-                string message = "Syntax Error: Unterminated string.";
-                return error UnterminatedStringError(message, line = self.currentLocation.line,
-                                                     column = self.currentLocation.column);
-            } else if (value == DOUBLE_QUOTE && previousChar != BACK_SLASH) {
+            if value == NEW_LINE {
+                lines.push(word);
+                word = "";
                 value = self.readNextChar();
-                break;
+            } else if value == DOUBLE_QUOTE {
+                previousChar = self.readNextChar();
+                value = self.charReader.peek();
+                if value == DOUBLE_QUOTE {
+                    previousChar = self.readNextChar();
+                    value = self.charReader.peek();
+                    if value == DOUBLE_QUOTE {
+                        value = self.readNextChar();
+                        lines.push(word);
+                        break;
+                    }
+                    word += previousChar;
+                }
+                word += previousChar;
             } else {
                 value = self.readNextChar();
                 word += value;
             }
             previousChar = value;
         }
-        return getToken(word, T_STRING, location);
+        word = self.getBlockStringValue(lines);
+        return getToken(word.trim(), T_STRING, location);
+    }
+
+    isolated function getBlockStringValue(string[] lines) returns string {
+        int? commonIndent = ();
+        string formatted = "";
+        foreach string line in lines {
+            int indent  = self.getLeadingWhiteSpaceCount(line);
+            if indent < line.length() && (commonIndent is () || indent < commonIndent) {
+                commonIndent = indent;
+                if commonIndent == 0 {
+                    break;
+                }
```

Comment:
Is this necessary? This is a special case right? Do we need to consider the special case here, since this will be handled anyways?

Reply:
will remove it